packages/buidler-core/src/internal/cli/cli.ts

```diff
@@ -89,11 +90,22 @@ async function main() {
     const ctx = BuidlerContext.createBuidlerContext();
     const config = loadConfigAndTasks(buidlerArguments.config);
 
+    const analytics = await Analytics.getInstance(
+      config.paths.root,
+      config.analytics.enabled
+    );
+
     const envExtenders = ctx.extendersManager.getExtenders();
     const taskDefinitions = ctx.tasksDSL.getTaskDefinitions();
 
     let taskName = parsedTaskName !== undefined ? parsedTaskName : "help";
 
+    // tslint:disable-next-line: prefer-const
+    let [abortAnalytics, hitPromise] = await analytics.sendTaskHit(taskName);
+
+    // tslint:disable-next-line: no-floating-promises
+    delay(300).then(() => (abortAnalytics = () => {}));
```

Comment:
I think this mail fail to work if the event loop is blocked by `compile` or another CPU intensive task. I can't be 100% sure about this, as it's a v8/node implementation detail, but would avoid it just in case.

I think doing something like this is more explicit, and doesn't rely on the event loop not being blocked:

```ts
const timestampBeforeRun = new Date().getTime();

await env.run(taskName, taskArguments);

const timestampAfterRun = new Date().getTime();

if (timestampAfterRun - timestampBeforeRun > ANALYTICS_SLOW_TASK_THRESHOLD) {
  await hitPromise;
} else {
  abortAnalytics();
}
```